backend/cmm_helpers.ml

```diff
@@ -4075,12 +4115,7 @@ let indirect_call ~dbg ty pos alloc_mode f args =
            ( Capply (ty, pos),
              [load ~dbg Word_int Asttypes.Mutable ~addr:(Cvar v); arg; Cvar v],
              dbg ))
-  | args ->
-    let arity = List.length args in
-    let l =
-      (Cconst_symbol (apply_function_sym arity alloc_mode, dbg) :: args) @ [f]
-    in
-    Cop (Capply (ty, pos), l, dbg)
+  | args -> call_caml_apply Asttypes.Mutable f args pos alloc_mode dbg
```

Comment:
It looks like `ty`'s use has been lost in 
this change, and the new code will
always use `typ_val`. Do we know that
`ty` is always `typ_val` at this point?

Reply:
This a bug I introduced in refactoring, fixing. Thanks!